core/src/main/java/com/sap/ai/sdk/core/AiCoreServiceWithDeployment.java

```diff
@@ -0,0 +1,156 @@
+package com.sap.ai.sdk.core;
+
+import com.sap.ai.sdk.core.client.DeploymentApi;
+import com.sap.ai.sdk.core.client.model.AiDeployment;
+import com.sap.cloud.sdk.cloudplatform.connectivity.DefaultHttpDestination;
+import com.sap.cloud.sdk.cloudplatform.connectivity.Destination;
+import com.sap.cloud.sdk.cloudplatform.connectivity.HttpDestination;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+
+/** Connectivity convenience methods for AI Core with deployment. */
+@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
+public class AiCoreServiceWithDeployment implements AiCoreServiceStub {
```

Comment:
```suggestion
public class AiCoreDeployment implements AiCoreServiceStub {
```

For me the `withDeployment` sounds a bit off, because there are many deployments on one AI Core service instance, and the semantic here is actually "withThisSingleDeploymentButWithoutAllTheOtherDeployments" 😄

Reply:
Sounds like, similar to [Charles idea](https://github.com/SAP/ai-sdk-java/pull/79#discussion_r1782649790), you would prefer the following method name changes, right?
```diff
- withDeployment[...]
+ forDeployment[...]
```